pkgs/development/libraries/libsvm/default.nix

```
@@ -1,33 +1,46 @@
-{lib, stdenv, fetchurl}:
+{ lib
+, stdenv
+, fetchurl
+, fixDarwinDylibNames
+, withOpenMP ? true
+}:
 
 stdenv.mkDerivation rec {
   pname = "libsvm";
-  version = "3.25";
+  version = "3.31";
 
   src = fetchurl {
     url = "https://www.csie.ntu.edu.tw/~cjlin/libsvm/libsvm-${version}.tar.gz";
-    sha256 = "sha256-UjUOiqdAsXbh13Pp3AjxNAIYw34BvsN6uQ2wEn5LteU=";
+    sha256 = "sha256-AKtWH0jfX8kqhCCa2P5Rmery5Rmzwnm6z8k1l4p1zx8=";
   };
 
-  buildPhase = ''
-    make
-    make lib
-  '';
-
-  installPhase = let
-    libSuff = stdenv.hostPlatform.extensions.sharedLibrary;
-  in ''
-    install -D libsvm.so.2 $out/lib/libsvm.2${libSuff}
-    ln -s $out/lib/libsvm.2${libSuff} $out/lib/libsvm${libSuff}
-    install -Dt $out/bin/ svm-scale svm-train svm-predict
-    install -Dm644 -t $out/include svm.h
-    mkdir $out/include/libsvm
-    ln -s $out/include/svm.h $out/include/libsvm/svm.h
-  '';
-
-  postFixup = lib.optionalString stdenv.isDarwin ''
-    install_name_tool -id libsvm.2.dylib $out/lib/libsvm.2.dylib;
-  '';
+  patches = lib.optional withOpenMP [ ./openmp.patch ];
```

SAGA disables OpenMP for Darwin:

```
pkgs/applications/gis/saga/default.nix:77:    "-DOpenMP_SUPPORT=${if stdenv.isDarwin then "OFF" else "ON"}"
```

some other packages do, too. I figure OpenMP must not be building on Darwin currently

```suggestion
  patches = lib.optional (!stdenv.isDarwin && withOpenMP) [ ./openmp.patch ];
```